packages/serverpod_test/lib/src/serverpod_test.dart

```diff
@@ -0,0 +1,415 @@
+import 'dart:async';
+
+import 'package:serverpod/serverpod.dart';
+import 'package:test/test.dart';
+
+export 'package:meta/meta.dart' show isTestGroup;
+
+/// A test specific session that is used to call database methods or pass to endpoints.
+abstract class TestSession implements DatabaseAccessor {
+  /// AuthenticationInfo for the session.
+  AuthenticationInfo? get authenticationInfo;
+
+  /// Creates a new unique session with the provided properties.
+  /// This is useful for setting up different session states in the tests
+  /// or simulating multiple users.
+  Future<TestSession> copyWith({
+    AuthenticationInfo? Function()? getAuthenticationInfo,
```

Comment:
💡 This is a workaround for to basically simulate `Optional<AuthenticationInfo?>. In other places in the code we have a `Undefined` type to solve this, but in this case a callback seemed more straight forward.